pkg/envoy/cds/builder.go

```diff
@@ -476,7 +476,8 @@ func (b *clusterBuilder) buildLocalClusters() []*xds_cluster.Cluster {
 	return clusters
 }
 
-func getTypedHTTPProtocolOptions(httpProtocolOptions *extensions_upstream_http.HttpProtocolOptions) (map[string]*any.Any, error) {
+// GetTypedHTTPProtocolOptions marshals the provided httpProtocolOptions into the consumable type for the envoy config.
+func GetTypedHTTPProtocolOptions(httpProtocolOptions *extensions_upstream_http.HttpProtocolOptions) (map[string]*any.Any, error) {
```

Comment:
It looks like this is only called from unexported functions? It feels strange to export it here, Is this part of #5109? I follow why  exporting the builders makes sense for future code, but not sure about these functions which seem like implementation details to the builders